camayoc/tests/qcs/utils.py

```diff
@@ -1,7 +1,54 @@
 """Utility functions for quipucords server tests."""
 
+from camayoc import api
+from camayoc.utils import uuid4
+from camayoc.qcs_models import HostCredential, NetworkProfile
+
 
 def assert_matches_server(qcsobject):
     """Assert that the data on the server matches this object."""
     other = qcsobject.read().json()
     assert qcsobject.equivalent(other)
+
+
+def assert_profile_update_fails(original_data, profile):
+    """Assert that the update method on this profile fails.
+
+    :param original_data: This should be the json you expect to match the
+        server. This can be collected from your object via profile.fields()
+        before altering the object with the invalid data.
+    """
+    # replace whatever client the profile had with one that won't raise
+    # exceptions
+    orig_client = profile.client
+    profile.client = api.Client(response_handler=api.echo_handler)
+    update_response = profile.update()
+    assert update_response.status_code in range(400, 500)
+    server_data = profile.read().json()
+    for key, value in server_data.items():
+        if key == 'credentials':
+            # the server creds are dicts with other data besides the id
+            cred_ids = []
+            for cred in value:
+                cred_ids.append(cred.get('id'))
+            assert sorted(original_data.get(key)) == sorted(cred_ids)
+        else:
+            assert original_data.get(key) == value
+    # give the profile its original client back
+    profile.client = orig_client
+
+
+def gen_valid_profile(cleanup, client, scanhost):
```

Comment:
scanhost is actually the hosts attribute of NetworkProfile maybe naming this hosts would be easier to figure out for what this parameter is for.

Reply:
This seems a factory function, we can make it generic by expecting `**kwargs` and passing it directly to the NetworkProfile constructor.

Reply:
Most of the calls to this function is passing api.Client() as the client, that signs to me that maybe it would be better to define client=None and if None create a new one, otherwise reuse the client passed.